cores/arduino/stm32/timer.c

```diff
@@ -1029,199 +1029,343 @@ void attachIntHandle(stimer_t *obj, void (*irqHandle)(stimer_t *))
 /*                            TIMx IRQ HANDLER                                */
 /******************************************************************************/
 
+#if defined(TIM1_BASE)
 /**
-  * @brief  TIM1 & TIM10 irq handler
+  * @brief  TIM1 IRQHandler
   * @param  None
   * @retval None
   */
+#if defined(STM32F0xx)
```

Comment:
Why not use TIM1_IRQHandler and define the proper one like the other (#define TIM1_IRQHandler TIM1_BRK_UP_TRG_COM_IRQHandler ...) in timer.h
in order to be homogeneous?

Reply:
I tried but when timer 10 or 16 are present, there is an error about redefinition:
'error: redefinition of 'TIM1_UP_TIM16_IRQHandler'

Reply:
On which MCU ?

however, for TIM10 handler, it seems strange for STM32F1xx
```
+#elif defined(STM32F2xx) || defined(STM32F4xx) || defined(STM32F7xx)
 void TIM1_UP_TIM10_IRQHandler(void)
```
but 
```
+#if defined(STM32F1xx) || defined(STM32F2xx) || defined(STM32F4xx) || defined(STM32F7xx)
   if(timer_handles[9] != NULL) {
     HAL_TIM_IRQHandler(timer_handles[9]);
```